src/main/java/com/example/employeePortal/controllers/EmployeeController.java

```diff
@@ -15,11 +18,11 @@
 public class EmployeeController {
     private final EmployeeService employeeService;
 
-    @GetMapping(value = "/by-name")
-    public List<Employee> findByName(@RequestParam("firstName") String firstName, @RequestParam("lastName") String lastName) {
-        List<Employee> employeesList = employeeService.findByName(firstName, lastName);
-        if(employeesList.isEmpty()){
-            throw new EmployeeNotFoundException("No employee found with the name: "+firstName+lastName);
+    @GetMapping(value = "/search")
+    public List<Employee> findByName(@RequestParam("query") String searchKeyword) {
+        List<Employee> employeesList = employeeService.findByName(searchKeyword);
+        if (employeesList.isEmpty()) {
+            throw new EmployeeNotFoundException("No employee found with the name containing: " + searchKeyword);
```

Comment:
No need to throw Exception matching employees not found. Simply return empty list.